src/main/java/teammates/logic/api/Logic.java

```diff
@@ -2622,4 +2622,14 @@ public MimeMessage emailErrorReport(String path, String params, Throwable error)
     @SuppressWarnings("unused")
     private void ____helper_methods________________________________________() {
     }
+    
+    public void extractActiveAndArchivedCourses(List<CourseDetailsBundle> courseBundles, 
```

Comment:
For now, let's keep it as two separate methods. You can add a TODO comment about optimizing it. The optimization is not directly relevant to the jstl conversion.